### PR TITLE
Unbundle Enhanced hearing CBM and Sound dampeners CBM

### PR DESCRIPTION
**Pull request by Cyrano7**
Unbundles Enhanced hearing CBM and Sound dampeners CBM. Adds Sound dampeners as a stand alone CBM. Changed description for both where relevant. For now Sound dampeners have same price and install difficulty as Enhanced hearing. Added Sound dampeners to same item groups as Enhanced hearing. Removed some code related to bundled install and uninstall. They should retain the same function and once both are installed activating one should deactivate the other as before. Will still be both included for the razorboy/girl profession. Let me know if I missed something. As for justification for the change, I think they serve pretty different functions and are used for different things.

### CODE REVIEW

**Comment by ejseto** [comment]
Another solution in search of a problem. A better question than, "Do these things serve similar function," is, "Are both of these strong enough to stand individually as a singular CBM?" We have enough "lockscreen widgets" imo. We ought to be *bundling* things like alarm clock and chronometer, rather than separating things that, quite frankly, *do* go together thematically. Why should we need half a dozen hearing aids cluttering things up?

**Comment by chaosvolt** [comment]
I'm not sure I like this idea.

The main gameplay reason that the sound dampeners are bundled with enhanced hearing is so you can get a good night's sleep. Though technically you can (or at least should be able to) just turn off the hearing enhancement, leaving it on makes any creature meandering around harmlessly liable to wake you up.

**Comment by Cyrano7** [comment]
@chaosvolt In the current build of the game you can just turn off enhanced hearing by itself. You can also turn on the sound dampeners which will automatically turn off enhanced hearing.

**Comment by chaosvolt** [comment]
True, that would work. That makes it viable to separate them, though not yet sure if it's worth doing.

**Comment by Cyrano7** [comment]
@ejseto A. Id say they are good enough to stand on there own. Enhanced hearing is about keeping the sounds around you within a certain range like an advanced hearing aid where the dampeners are more of earplugs, good for sleeping. B. What lockscreen widgets are you referring to? C. I don't think we need a ton of hearing aids in the game either. Currently we have one hearing related CBM, noise canceling headgear (pretty much makeshift earplugs) and normal pre-cataclysm earplugs. I'm pretty sure there is also a helmet that serves the same function as advanced hearing as well.  But that's only 4 sound protection related items currently.

File: data/json/item_groups.json [hunk 1]
```diff
@@ -6560,6 +6560,7 @@
       ["bio_painkiller", 10],
       ["bio_alarm", 10],
       ["bio_ears", 10],
+	  ["bio_earplugs", 10],
```
Comment (illi-kun) [inline data/json/item_groups.json]: You're using tabs here. That's definitely not good, please replace them here and below by decent amount of whitespaces.

**Comment by Cyrano7** [comment]
@illi-kun Thanks for catching that.

**Comment by kevingranade** [comment]
The problem with, "I think they serve different functions" is it's very
subjective, obviously the original author thought they were related, and
several commentors agree, I'd be looking for a much stronger benefit for
the change before saddling ourselves with managing a migration of existing
content like this.
On that subject, what happens to existing saves that have enhanced hearing
installed? We shouldn't take away functionality they already have.

**Comment by Cyrano7** [comment]
@kevingranade On the subject of existing saves. the changes I've made don't really effect the installed bionics. Just the items, install, uninstall. On my expanded reason for the change I'm responded to ejseto "Id say they are good enough to stand on there own. Enhanced hearing is about keeping the sounds around you within a certain range like an advanced hearing aid where the dampeners are more of earplugs, good for sleeping." To expanded bit further it's a bit of a balance thing I think. Getting three different functions however minor they for the price of one CBM install is a lot of bang for your buck. Thanks for getting back to me though even if its not the right fit for the game.

File: data/json/item_groups.json [hunk 2]
```diff
@@ -6560,6 +6560,7 @@
       ["bio_painkiller", 10],
       ["bio_alarm", 10],
       ["bio_ears", 10],
+	   ["bio_earplugs", 10],
```
Comment (illi-kun) [inline data/json/item_groups.json]: Uh, you actually didn't solve the issue (see below), but it seems that it's better to focus on the general discussion of this PR before you apply any more efforts to solving this (actually minor) 'TAB-to-space' issue.
untitled

**Comment by ejseto** [comment]
I don't see the problem with getting a lot of bang for your buck. Items are balanced both around power and frequency of acquisition. CBMs are powerful. They're supposed to be. Do we really want to fill up the game with a bunch of CBM wristwatches, CBM earplugs, CBM thermometers, and so on? The flip side of balancing the power of CBMs is they become more trouble than they are worth. Why should I risk getting two defective CBMs when I can just wear a diver's watch that tells time AND temperature?

**Comment by narc0tiq** [comment]
To the best of my knowledge, originally these were a single CBM that got split to allow disabling each of its two aspects independently of the other. As far as I'm concerned, you aren't installing two bionics but a single configurable ear enhancement.

**Comment by ejseto** [comment]
>single configurable ear enhancement

Exactly. There's no way to make a single 3-state switching CBM without adding a player class variable. CBMs are either on or off. So it got split into two CBMs with 4 possible switched values (with the contradictory "both on" thrown away).

**Comment by Cyrano7** [comment]
@ejseto "Exactly. There's no way to make a single 3-state switching CBM without adding a player class variable. CBMs are either on or off. So it got split into two CBMs with 4 possible switched values (with the contradictory "both on" thrown away)." This changes my thinking a bit honestly. I'm not as "zealous" as you think. I can still be talked away form my opinions sometimes. There is merit in your arguments.

This is off topic but after reading "single configurable ear enhancement" and "CBMs with 4 possible switched values" gave me idea. I could add a blindfold type CBM (far to weak of a CBM to stand on its own) where you could toggle your vision bundled with the anti-glare CBM. It would be basically the eye version of the enhanced hearing bundle and along the same thinking. blindfold+sound dampeners+alarm system+ would be pretty cool, doesn't really effect the balance of the game. I know you probably wont like it though. Just another "widget".

**Comment by ejseto** [comment]
I'm not against "widgets." I just don't think they're strong enough to stand on their own. Chronometer and weather reader are the obvious examples. But even if they have almost no value since I can just wear a diver's watch, they have at minimum zero value. Adding content has that virtue. As long as it doesn't interfere with the rest of the game, it has at minimum zero value. If I don't like it, I don't have to install it. Changing things doesn't benefit from that. You could make it better. But you could also make it worse. There's a reason they say, "If it ain't broke, don't fix it."

**Comment by Cyrano7** [comment]
Off Topic: I think chronometer/alarm is useful enough. Weather reader is pretty weak alone. I don't know if there is something to bundle it with though. As for my idea is there any yay/nay's out there from people still paying attention. Anti-Glare CBM bundle. The name Optical Dampeners work? Shouldn't be to difficult to do.

On Topic: As there are currently no votes for yes i'm going to close this soon.

**Comment by kevingranade** [comment]
No problem with the damper thing.  It is useful in the same way that
turning off hearing is, sleeping when stimuli would otherwise wake tipu up.

**Pull request closed without merging**